### PR TITLE
Add analyzer support for status code methods and constructors

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @khellang

File: src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs [hunk 1]
```diff
@@ -29,6 +29,7 @@ public ObjectResult(object value)
         /// <summary>
         /// Gets or sets the HTTP status code.
         /// </summary>
+        [StatusCodeValue]
         public int? StatusCode { get; set; }
```
Comment (khellang) [inline src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs]: What about the other properties that implement `IStatusCodeActionResult`? Wouldn't it be nicer to check if a property implements the interface instead?

**Comment by khellang** [comment]
This looks good. Only one comment from me 😊

File: src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs (hunk 1 above)
Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs]: Done. Switched to using the interface contract to compare, forgot all about it

**Pull request closed without merging**